source/platforms/GitLab.ts

```diff
@@ -99,28 +100,49 @@ class GitLab implements Platform {
   updateOrCreateComment = async (dangerID: string, newComment: string): Promise<string> => {
     d("updateOrCreateComment", { dangerID, newComment })
 
-    //Even when we don't set danger to create threads, we still need to delete them if someone answered to a single
-    // comment created by danger, resulting in a discussion/thread. Otherwise we are left with dangling comments
-    // that will most likely have no meaning out of context.
-    const discussions = await this.getDangerDiscussions(dangerID)
-    const firstDiscussion = discussions.shift()
-    const existingNote = firstDiscussion?.notes[0]
-    // Delete all notes from all other danger discussions (discussions cannot be deleted as a whole):
-    await this.deleteNotes(this.reduceNotesFromDiscussions(discussions)) //delete the rest
-
-    let newOrUpdatedNote: GitLabNote
+    if (useThreads()) {
```

Comment:
The main difference in the approach in this pull request is to treat the new "create a thread for the main comment" behavior as completely separate from the previous behavior to "create a note for the main comment".

Mixing the logic of these two behaviors caused issues where certain scenarios were not handled, like trying to update the "main" Danger comment and instead updating an inline comment because it was a discussion and the "main" Danger comment was not.